src/bcdiceCore.rb

```diff
@@ -993,18 +998,76 @@ def rand(max)
     return value
   end
 
+  DetailedRandResult = Struct.new(:kind, :sides, :value)
+
+  # @params [Integer] max
+  # @return [Integer] 0以上max未満の整数
+  def rand(max)
+    ret = rand_inner(max)
+
+    push_to_detail(:nomal, max, ret + 1)
+    return ret
+  end
+
+  # 十の位をd10を使って決定するためのダイスロール
+  # @return [Integer] 0以上90以下で10の倍数となる整数
+  def roll_tens_d10()
+    # rand_innerの戻り値を10倍すればすむ話なのだが、既存のテストとの互換性の為に処理をする
+    r = rand_inner(10) + 1
+    if r == 10
+      r = 0
+    end
+
+    ret = r * 10
+
+    push_to_detail(:tens_d10, 10, ret)
+    return ret
+  end
+
+  # d10を0~9として扱うダイスロール
+  # @return [Integer] 0以上9以下の整数
+  def roll_d9()
+    ret = rand_inner(10)
+
+    push_to_detail(:d9, 10, ret)
+    return ret
+  end
+
   def setCollectRandResult(b)
     if b
       @randResults = []
+      @detailedRandResults = []
     else
       @randResults = nil
+      @detailedRandResults = nil
     end
   end
 
+  # @params [Symbol] kind
+  # @params [Integer] sides
+  # @params [Integer] value
+  def push_to_detail(kind, sides, value)
+    unless @detailedRandResults.nil?
+      detail = DetailedRandResult.new(kind, sides, value)
+      @detailedRandResults.push(detail)
+    end
+  end
+
+  # @return [Array<Array<(Integer, Integer)>>]
```

Comment:
YARDのコメントについて、`(Integer, Integer)` で `[整数, 整数]` という配列を表すようです（`Array(Integer, Integer)` の省略形、下の画像を参照）。

```suggestion
  # @return [Array<(Integer, Integer)>]
```

https://yardoc.org/types.html

<img width="983" alt="yard-type-array" src="https://user-images.githubusercontent.com/2551173/76704334-9cc2a000-671b-11ea-9585-d036fa39080c.png">